include/deal.II/lac/lapack_full_matrix.h

```diff
@@ -420,6 +420,42 @@ class LAPACKFullMatrix : public TransposeTable<number>
                const LAPACKFullMatrix<number> &B,
                const bool                      adding=false) const;
 
+  /**
+   * Matrix-matrix-multiplication using transpose of <tt>this</tt> and a
+   * diagonal vector @p V.
+   *
+   * If the <code>adding=false</code> then the result is stored in the matrix
+   * $C = A^T \rm{diag}(V) B$
+   * otherwise it is added $C \mathrel{+}= A^T \rm{diag}(V) B$.
+   *
+   * @note It is assumed that @p A, @p B and @p V have compatible sizes and that
+   * @p C already has the right size.
+   *
+   * @note This function is not provided by LAPACK. The function first forms $BV$ product and
+   * then uses Xgemm function.
+   */
+  void Tmmult (LAPACKFullMatrix<number>       &C,
+               const LAPACKFullMatrix<number> &B,
+               const Vector<number>           &V,
+               const bool                      adding=false) const;
+
+  /**
+   * Matrix-matrix-multiplication using transpose of <tt>this</tt> and a
+   * diagonal vector @p V. Similar to the above but computes $C = A^T \rm{diag}(V)^2 A$
+   * when <code>adding=false</code> and $C \mathrel{+}= A^T \rm{diag}(V)^2 A$
+   * otherwise. If @p V is empty, it's taken as identity matrix which corresponds
+   * to calculation of $A^T A$.
+   *
+   * @note It is assumed that @p A and @p V have compatible sizes and that
+   * @p C already has the right size.
+   *
+   * @note This function is not provided by LAPACK. The function first forms $VA$ product and
+   * then uses Xsyrk function.
+   */
+  void Tmmult (LAPACKFullMatrix<number>       &C,
```

Comment:
@kronbichler 

an alternative solution: one could add `scale_rows(const Vector<number> &V)` to get `diag(V)*A`, then `Tmmult()` will be clever enough to do `dsyrk` instead of `gemm`.
So the user would need to do:
```
B=A;
B.scale_rows(V);
B.Tmmult(C,B);
```
instead of
```
A.Tmmult(C,V);
```

That's an option.

Reply:
p.s. manual `scale_rows` is being added to ScaLAPACK here https://github.com/dealii/dealii/pull/5869